java/client/src/test/java/glide/api/RedisClusterClientTest.java

```diff
@@ -1,101 +1,165 @@
+        // setup
+        String key = "testKey";
+        String cmd = "GETSTRING";
+        String[] arguments = new String[] {cmd, key};
+        CompletableFuture<ClusterValue<Object>> testResponse = mock(CompletableFuture.class);
+        InterruptedException interruptedException = new InterruptedException();
+        when(testResponse.get()).thenThrow(interruptedException);
+        when(commandManager.<ClusterValue<Object>>submitNewCommand(any(), any(), any(), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        InterruptedException exception =
+                assertThrows(
+                        InterruptedException.class,
+                        () -> {
+                            CompletableFuture<ClusterValue<Object>> response = service.customCommand(arguments);
+                            response.get();
+                        });
+
+        // verify
+        assertEquals(interruptedException, exception);
     }
 
     @Test
-    @SneakyThrows
-    public void custom_command_with_multi_node_route_returns_multi_value() {
-        var commandManager = new TestCommandManager(null);
-
-        var data = Map.of("key1", "value1", "key2", "value2");
-        var client = new TestClient(commandManager, data);
-
-        var value = client.customCommand(new String[0], () -> false).get();
-        assertAll(
-                () -> assertTrue(value.hasMultiData()), () -> assertEquals(data, value.getMultiValue()));
+    public void info_success() throws ExecutionException, InterruptedException {
+        // setup
+
+        CompletableFuture<ClusterValue<Map>> testResponse = mock(CompletableFuture.class);
+        Map testPayload = new HashMap<String, String>();
+        testPayload.put("key1", "value1");
+        testPayload.put("key2", "value2");
+        testPayload.put("key3", "value3");
+        when(testResponse.get()).thenReturn(ClusterValue.of(testPayload));
+        when(commandManager.<ClusterValue<Map>>submitNewCommand(any(), any(), any(), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<ClusterValue<Map>> response = service.info();
+
+        // verify
+        assertEquals(testResponse, response);
+        ClusterValue<Map> clusterValue = response.get();
+        assertTrue(clusterValue.hasMultiData());
+        Map payload = clusterValue.getMultiValue();
+        assertEquals(testPayload, payload);
     }
 
-    private static class TestClient extends RedisClusterClient {
-
-        private final Object object;
-
-        public TestClient(CommandManager commandManager, Object objectToReturn) {
-            super(null, commandManager);
-            object = objectToReturn;
-        }
-
-        @Override
-        protected Object handleObjectResponse(Response response) {
-            return object;
-        }
+    @Test
+    public void info_withRoute_success() throws ExecutionException, InterruptedException {
```

Comment:
Consider renaming all tests and moving to the separate file(s)
```suggestion
    @SneakyThrows
    public void info_withRoute_success() {
```